packages/server/src/db/db.service.ts

```diff
@@ -302,40 +311,107 @@ export default class DBService {
     pictureUrls: string[]
   ) {
     const shelf = await this.shelfRepository.findOne(shelfId);
-    return this.flowerRepository.update(
-      id,
-      { shelf, name, description, order, rrules, pictureUrls },
-    );
+    return this.flowerRepository.update(id, {
+      shelf,
+      name,
+      description,
+      order,
+      rrules,
+      pictureUrls,
+    });
   }
 
   deleteFlower(id: number) {
     return this.flowerRepository.delete(id);
   }
 
+  @Transaction()
+  private async moveFlowerInTransaction(
+    flowerId: number,
+    shelfId: number,
+    targetShelfId: number,
+    @TransactionManager() entityManager: EntityManager
+  ) {
+    const flowerRepInTransaction = entityManager.getRepository(FlowerEntity);
+    const shelfRepInTransaction = entityManager.getRepository(ShelfEntity);
+    const findShelfConfig = {
+      relations: ['flowers'],
+    };
+    const findFlowerConfig = {
+      relations: ['shelf'],
+    };
+    const movingFlower = await flowerRepInTransaction.findOne(
+      flowerId,
+      findFlowerConfig
+    );
+    const shelf = await shelfRepInTransaction.findOne(shelfId, findShelfConfig);
+    const targetShelf = await shelfRepInTransaction.findOne(
+      targetShelfId,
+      findShelfConfig
+    );
+    const updatedFlower = { ...movingFlower, shelf: targetShelf };
+    const updatedOldShelf = {
+      ...shelf,
+      flowers: shelf.flowers.filter(({ id }) => id !== movingFlower.id),
+    };
+    const updatedTargetShelf = {
+      ...targetShelf,
+      flowers: [...targetShelf.flowers, movingFlower],
+    };
+    await flowerRepInTransaction.update(flowerId, updatedFlower);
+    return {
+      flower: updatedFlower,
+      shelf: updatedOldShelf,
+      targetShelf: updatedTargetShelf,
+    };
+  }
+
+  async moveFlower(flowerId: number, shelfId: number, targetShelfId: number) {
+    return this.moveFlowerInTransaction(
+      flowerId,
+      shelfId,
+      targetShelfId,
+      this.entityManager
+    );
+  }
+
   async getUsersByFlowerId(id: number) {
-    const { shelf: { id: shelfId } } = await this.flowerRepository.findOne(id, { relations: ['shelf'] });
-    const userToShelves = await this.userToShelfRepository.find({ where: { shelfId }, relations: ['user'] });
-    return userToShelves.map(u2s => u2s.user);
+    const {
+      shelf: { id: shelfId },
+    } = await this.flowerRepository.findOne(id, { relations: ['shelf'] });
+    const userToShelves = await this.userToShelfRepository.find({
+      where: { shelfId },
+      relations: ['user'],
+    });
+    return userToShelves.map((u2s) => u2s.user);
   }
 
-  async saveAction(userId: number, flowerId: number, action: Actions, timestamp: number) {
+  async saveAction(
+    userId: number,
+    flowerId: number,
+    action: Actions,
+    timestamp: number
```

Comment:
There's no eslint rule enforcing this. Why does it still get changed? Are you changing these manually?